From d7aa275b0bca86ae174e7e504dd269df2a0234cf Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Tue, 6 Apr 2021 09:56:55 +0300
Subject: [PATCH] Export SplashFont* symbols used by Scribus

Scribus 1.5.6.1 compiled against poppler 21.04.0 was unable to load its
PDF importer plugin without these classes being exported by libpoppler:

- SplashFontFileID
- SplashFontEngine
- SplashFontSrc
---
 splash/SplashFontEngine.h | 3 ++-
 splash/SplashFontFile.h   | 3 ++-
 splash/SplashFontFileID.h | 4 +++-
 3 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/splash/SplashFontEngine.h b/splash/SplashFontEngine.h
index becc0d9ce..3e5458c09 100644
--- a/splash/SplashFontEngine.h
+++ b/splash/SplashFontEngine.h
@@ -30,6 +30,7 @@
 #include <array>
 
 #include "SplashTypes.h"
+#include "poppler_private_export.h"
 
 class SplashT1FontEngine;
 class SplashFTFontEngine;
@@ -44,7 +45,7 @@ class SplashFontSrc;
 // SplashFontEngine
 //------------------------------------------------------------------------
 
-class SplashFontEngine
+class POPPLER_PRIVATE_EXPORT SplashFontEngine
 {
 public:
     // Create a font engine.
diff --git a/splash/SplashFontFile.h b/splash/SplashFontFile.h
index 598fd8325..539bd3075 100644
--- a/splash/SplashFontFile.h
+++ b/splash/SplashFontFile.h
@@ -23,6 +23,7 @@
 #define SPLASHFONTFILE_H
 
 #include "SplashTypes.h"
+#include "poppler_private_export.h"
 
 class GooString;
 class SplashFontEngine;
@@ -33,7 +34,7 @@ class SplashFontFileID;
 // SplashFontFile
 //------------------------------------------------------------------------
 
-class SplashFontSrc
+class POPPLER_PRIVATE_EXPORT SplashFontSrc
 {
 public:
     SplashFontSrc();
diff --git a/splash/SplashFontFileID.h b/splash/SplashFontFileID.h
index 02b528aab..c127dd87f 100644
--- a/splash/SplashFontFileID.h
+++ b/splash/SplashFontFileID.h
@@ -21,11 +21,13 @@
 #ifndef SPLASHFONTFILEID_H
 #define SPLASHFONTFILEID_H
 
+#include "poppler_private_export.h"
+
 //------------------------------------------------------------------------
 // SplashFontFileID
 //------------------------------------------------------------------------
 
-class SplashFontFileID
+class POPPLER_PRIVATE_EXPORT SplashFontFileID
 {
 public:
     SplashFontFileID();
-- 
GitLab

